File: Challenge139E.py
# Tui Popenoe
# Challenge139E.py - Pangrams

"""Returns true if the given string is a pangram, i.e. contains every letter 
    in the given language"""

def pangram(string=None, language=None):
    if string is None:
        print('Enter string: ')
        string = raw_input()

    if language is None:
        language = english = ['a','b','c','d','e','f','g','h','i','j','k','l',\
            'm','n','o','p','q','r','s','t','u','v','w','x','y','z']

    for i in range(len(string)):
        for j in range(len(language)):
            if string[i].lower() == language[j]:
                language.pop(j)
                break

    # print('language: ')
    # print(language)
    # All the letters have been popped from the list
    if not language:
        print('True')
        return True
    else:
        print('False')
        return False

def main():
    pangram()

if __name__ == '__main__':
    main()

File: Challenge139E_Pangrams/challenge139E.cpp
// C++ 11.0
// Tui Popenoe
// Challenge139E.cpp - Pangrams

# include <iostream>
# include <string>
# include <utility>

bool pangram(string, string);

int main(){

    std::string alphabet = {'a','b','c','d','e','f','g','h','i','j','k','l',
                       'm','n','o','p','q','r','s','t','u','v','w','x','y','z'}


    return 0;
}

bool pangram(string input, string alphabet){

    std::transform(input.begin(), input.end(), input.begin(), ::tolower);
    for(int i = 0; i < input.length() -1; i++){
        for(int j = 0; j < language.length() -1; j++){
            if(input[i] == language[j]){
                
            }
        }
    }
}